cmd/ops_agent_windows/run_windows.go

```diff
@@ -163,6 +162,51 @@ func (s *service) startSubagents() error {
 	return nil
 }
 
+func (s *service) SendMetricsEveryIntervalWindows(metrics []self_metrics.IntervalMetrics, r <-chan svc.ChangeRequest, changes chan<- svc.Status) error {
+	bufferChannel := make(chan []self_metrics.Metric)
+	buffer := make([]self_metrics.Metric, 0)
+
+	tickers := make([]*time.Ticker, 0)
+
+	for _, m := range metrics {
+		tickers = append(tickers, time.NewTicker(time.Duration(m.Interval)*time.Minute))
+	}
+
+	for idx, m := range metrics {
+		go self_metrics.RegisterMetric(m, bufferChannel, tickers[idx])
+	}
+	s.log.Info(1, "registered metric tickers")
+
+	defer func() {
```

Comment:
Why did you need to pull this into the function? You could create an independent goroutine that watches the ticker (and probably another channel for the exit signal).

Reply:
Yes, if I only pass the channel `changes` as an argument this is not needed to be inside the function. The real reason was mirroring what was already implemented in the current `Execute` implementation. This is probably linked to make a unified `SendMetricsEveryInterval` implementation for both platforms.